### PR TITLE
Updates Benchmark macro parsing to use Generic Argument

### PR DESCRIPTION
**Pull request by gupnik**
Uses Oliver's suggestion as discussed in #13897

Will update #13845 once this is merged.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2683152

**Review by sam0x17 — COMMENTED** [review]
This looks great to me now that I know you can do this with the `{}` generic syntax -- let's get some tests

File: frame/support/procedural/src/benchmark.rs [hunk 1]
```diff
@@ -228,17 +228,8 @@ fn parse_params(item_fn: &ItemFn) -> Result<Vec<ParamDef>> {
...
 		let Ok(args) = syn::parse::<RangeArgs>(args) else { return invalid_param(typ.span()) };
-		let Ok(start) = args.start.base10_parse::<u32>() else { return invalid_param(args.start.span()) };
-		let Ok(end) = args.end.base10_parse::<u32>() else { return invalid_param(args.end.span()) };
 
-		if end < start {
-			return Err(Error::new(
-				args.start.span(),
-				"The start of a `ParamRange` must be less than or equal to the end",
-			))
-		}
```
Comment (sam0x17) [inline frame/support/procedural/src/benchmark.rs:239]: we can probably replace this with an assertion at runtime

Comment (gupnik) [reply frame/support/procedural/src/benchmark.rs:239]: Added a check.

File: frame/support/test/tests/benchmark_ui/bad_param_range.stderr [hunk 2]
```diff
@@ -1,5 +1,7 @@
-error: The start of a `ParamRange` must be less than or equal to the end
-  --> tests/benchmark_ui/bad_param_range.rs:10:21
-   |
-10 |     fn bench(x: Linear<3, 1>) {
-   |                        ^
+error[E0080]: evaluation of constant value failed
+ --> tests/benchmark_ui/bad_param_range.rs:5:1
+  |
+5 | #[benchmarks]
+  | ^^^^^^^^^^^^^ attempt to compute `0_usize - 1_usize`, which would overflow
```
Comment (ggwpez) [inline frame/support/test/tests/benchmark_ui/bad_param_range.stderr:5]: Yea not sure how helpful this error is. Maybe we just do it at runtime?

**Review by ggwpez — APPROVED** [review]
% that error message

**Comment by gupnik** [comment]
bot fmt

File: frame/support/test/tests/benchmark_ui/bad_param_range.stderr (hunk 2 above)
Comment (gupnik) [reply frame/support/test/tests/benchmark_ui/bad_param_range.stderr]: Added a runtime check.

**Comment by gupnik** [comment]
bot fmt

**Review by sam0x17 — APPROVED** [review]

**Comment by gupnik** [comment]
bot merge

**Pull request merged**